components/authz-service/storage/postgres/rule_test.go

```diff
@@ -131,6 +131,23 @@ func TestCreateRule(t *testing.T) {
 			assertCount(t, 3, db.QueryRow(
 				`SELECT count(*) FROM iam_staged_rule_conditions WHERE rule_db_id=(SELECT r.db_id FROM iam_staged_project_rules r WHERE r.id=$1)`, ruleID))
 		},
+		"when the project filter does not match, return ErrNotFound": func(t *testing.T) {
```

Comment:
I wonder, given the confusion yesterday, if we should also have  "if no projects allowed" and "if all projects allowed" test cases?

Reply:
Added test for "all". For the "no projects allowed case", it would never get here--that is caught earlier with this error:
```
"unauthorized: members [\"token:t1\"] cannot perform action \"iam:projects:update\" on resource \"iam:projects:proj2\" filtered by projects []",
```